app/src/main/java/org/glucosio/android/tools/ReadingToCSV.java

```diff
@@ -30,22 +30,15 @@
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.OutputStreamWriter;
-import java.util.ArrayList;
+import java.util.List;
 
 import io.realm.Realm;
 
 public class ReadingToCSV {
 
-    private Context context;
-
-    public ReadingToCSV(Context mContext) {
-        this.context = mContext;
-    }
-
-
-    public String createCSVFile(Realm realm, final ArrayList<GlucoseReading> readings, String um) {
+    public static String createCSVFile(Context context, Realm realm, final List<GlucoseReading> readings, String um) {
```

Comment:
Clear!

I would not bother about performance unless you show me it is the bottle neck or some problem here. Because this again breaks testability. With this solution, I can not mock this behaviour in unit tests. So if I test any class that uses this functionality, I will also test real exporting as well.

An article about "love" to statics http://www.endran.nl/blog/death-to-statics/